core/trino-main/src/main/java/io/trino/operator/scalar/ArrayUnionFunction.java

```diff
@@ -61,22 +61,24 @@ public static Block union(
     {
         int leftArrayCount = leftArray.getPositionCount();
         int rightArrayCount = rightArray.getPositionCount();
-        TypedSet typedSet = createEqualityTypedSet(type, elementEqual, elementHashCode, leftArrayCount + rightArrayCount, "array_union");
         BlockBuilder distinctElementBlockBuilder = type.createBlockBuilder(null, leftArrayCount + rightArrayCount);
-        appendTypedArray(leftArray, type, typedSet, distinctElementBlockBuilder);
-        appendTypedArray(rightArray, type, typedSet, distinctElementBlockBuilder);
+        TypedSet typedSet = createEqualityTypedSet(
+                type,
+                elementEqual,
+                elementHashCode,
+                distinctElementBlockBuilder,
```

Comment:
Can this impact the order of elements returned from the function? Do we want to tie TypeSet internals to maintain ordering?
Move this commit to a separate PR please, so it can be taken a closer look at.

Reply:
`TypedSet` does maintain insertion order, keeping the first copy of each element added (because it never modifies its block builder besides appending new elements).

I'll split the PR shortly.

Reply:
Split as #9905.